docs/tutorials/SignalsPrimer.m

```diff
@@ -354,18 +373,17 @@
   true, c);
 
 %% Demonstration of scan
-% Scan is a very powerful method that allows one to map a signal's current
-% value and it's previous value through a function.  This allows one to
-% define Signals that have some sort of history to them. In other
-% functional programming applications this may be called fold or reduce.
```

Comment:
I don't think this is really accurate, because in my experience 'fold' and 'reduce' always compress the output of 'scan' into a single value. E.g. If you run 'scan' on an array, the entire array is returned, but 'fold' or 'reduce' will return a single value.

Reply:
See Chris's Signals presentation and [this post](https://medium.com/@atnemecek/the-scan-function-is-the-core-of-all-programming-5f5085df76e7).  Also note the comparison between the two [here](https://reactive.how/reduce).

The output of scan depends on the function.  Many applications of the scan method return scalar values, not arrays.  e.g. `y = arr.scan(@(a,b) sum([a b]), 0)`, or `max = arr.scan(@(prev,new) max([prev new]), nan)`.

The comparison is useful because it describes the purpose of `scan` to users new to functional reactive programming who are nonetheless familiar with normal functional programming. Please leave it in, but it should be reworded thus:
```
% Scan is a very powerful method that allows one to map a signal's current
% value and it's previous value through a function.  This allows one to
% define Signals that have some sort of history to them. This is essentially 
% the same as the fold or reduce functions found in other functional 
% programming applications.
```

Reply:
hm yes that's true about scan's ability to return scalars depending on it's input function, but still 'scan' and 'fold/reduce' are different (in terms of the former returning the accumulator piece-wise, and the latter returning it at the end of the total accumulation). 'Scan', like that post I've seen before mentions, is at the core of functional programming, and I don't know of a language that emphasizes functional programming that doesn't have separate 'scan' and 'fold/reduce' functions, so I don't think the comparison is necessary for the audience you describe here,
```The comparison is useful because it describes the purpose of scan to users new to functional reactive programming who are nonetheless familiar with normal functional programming.``` 
unless there's a way that you see scan acting differently in functional as opposed to functional reactive? If not, what about changing the last sentence in the comment to:
```The final output of scan is the output of fold or reduce functions...```

Reply:
To my knowledge `scan` is not a core function in the same way as `map`, `fold` and `reduce`.  That is, you don't find `scan` in all functional programming languages, it's far less common.  Hence, it's worth quickly noting that it's practically very similar to the latter.  How about we just say they're similar and leave it at that:

```
% Scan is a very powerful method that allows one to map a signal's current
% value and it's previous value through a function.  This allows one to
% define Signals that have some sort of history to them. This is similar to
% the fold or reduce functions found in other functional programming 
% applications.
```

Reply:
ok, out of curiosity do you have examples of languages it's not implemented in or other reasons why you think it's less common? I've only worked with racket, sml, and haskell as FP-oriented languages, and all of them have scan, and Ruby does as well, which led me to believe it's common.

Reply:
This is outside of my area of expertise but JavaScript and Python are two examples of languages that have those three built in but not `scan`.  From what I've seen `map`, `fold`/`reduce` and `filter` are always mentioned together in articles and tutorials (e.g. [1](http://www.cse.unsw.edu.au/~en1000/haskell/hof.html), [2](https://medium.com/@eliaai/javascript-functional-programming-foreach-map-filter-reduce-f710abb69753), [3](https://www.oracle.com/technetwork/community/join/member-discounts/523385), [4](https://coldfusion.adobe.com/2017/10/map-reduce-and-filter-functions-in-coldfusion/), [5](https://web.mit.edu/6.005/www/fa15/classes/25-map-filter-reduce/)) whereas you have to dig much deeper to find mentions of `scan`.  Wikipedia has devoted pages for [map](https://en.wikipedia.org/wiki/Map_(higher-order_function)), [fold ](https://en.wikipedia.org/wiki/Fold_(higher-order_function)) and [filter](https://en.wikipedia.org/wiki/Filter_(higher-order_function)), but [not scan](https://en.wikipedia.org/wiki/Scan_(higher-order_function)).

If you Google "functional programming x" where x is:
`map`: 61,200,000 results
`reduce`: 54,900,000 results
`filter`: 38,800,000 results
`scan`: 19,700,000 results

Scan isn't mentioned in many common reference pages or 'cheat sheets', whereas the others are.  Some random examples: [Racket](https://docs.racket-lang.org/racket-cheat/index.html?q=cheatsheet), [Haskell](http://www.cse.unsw.edu.au/~en1000/haskell/inbuilt.html), [Lisp](http://jtra.cz/stuff/lisp/sclr/index.html).

Reply:
ok, I've seen 'scan' mentioned in some 'cheat sheets', but fair enough; the "big three" definitely seem to be 'map, 'filter' 'reduce'